lib/options.js

```diff
@@ -12,90 +12,104 @@ module.exports = {
    * availableOptions - allowed values (only for type=enum)
    * description - human-readable description of the item
    * external - whether the option is settable via the API
+   * usage - array of supported types of usage (i.e. CONVERSION, VALIDATION)
    *
    * @param {string} [mode='document'] Describes use-case. 'document' will return an array
    * with all options being described. 'use' will return the default values of all options
+   * @param {Object} criteria Decribes required criteria for options to be returned. can have properties
+   *   external: <boolean>
+   *   usage: <array> (Array of supported usage type - CONVERSION, VALIDATION)
    * @returns {mixed} An array or object (depending on mode) that describes available options
    */
-  getOptions: function(mode = 'document') {
+  getOptions: function(mode = 'document', criteria = {}) {
+    // Override mode & criteria if first arg is criteria (objects)
+    if (typeof mode === 'object') {
+      criteria = mode;
+      mode = 'document';
+    }
+
     let optsArray = [
       {
-        name: 'Set request name source',
+        name: 'Naming requests',
         id: 'requestNameSource',
         type: 'enum',
-        default: 'fallback',
-        availableOptions: ['url', 'fallback'],
-        description: 'Option for setting source for a request name',
-        external: true
+        default: 'Fallback',
+        availableOptions: ['Url', 'Fallback'],
+        description: 'Determines how the requests inside the generated collection will be named.' +
+        ' If “Fallback” is selected, the request will be named after one of the following schema' +
```

Comment:
not sure if we require `following` here

Reply:
It's what's mentioned in Importer Setting doc. We can carry this discussion over there.